test/libponyc/annotations.cc

```diff
@@ -187,3 +187,27 @@ TEST_F(AnnotationsTest, InternalAnnotation)
 
   TEST_ERROR(src);
 }
+
+TEST_F(AnnotationsTest, StandardAnnotationLocationGood)
+{
+  const char* src =
+    "struct \\packed\\ Foo\n"
+    "  fun foo() =>\n"
+    "    if \\likely\\ bar then None end\n"
+    "    while \\unlikely\\ bar do None end\n"
+    "    repeat None until \\likely\\ bar end\n"
+    "    match bar | \\unlikely\\ bar => None end";
+
+  TEST_COMPILE(src, "syntax");
+}
+
+TEST_F(AnnotationsTest, StandardAnnotationLocationBad)
+{
+  const char* src =
+    "class \\packed\\ Foo\n"
+    "  fun foo() =>\n"
+    "    try \\likely\\ bar else None end\n"
+    "    repeat \\likely\\ None until \\likely\\ bar end";
+
+  TEST_ERROR(src);
```

Comment:
Would be nice to use the `TEST_ERRORS_{N}` macros here to make sure we're erroring on every bad case, instead of just one of them.

However, I wouldn't block this PR over that detail.

Reply:
I'll update.